base/mysqlxx/PoolWithFailover.cpp

```diff
@@ -136,6 +136,21 @@ PoolWithFailover::Entry PoolWithFailover::get()
 
                     if (!entry.isNull())
                     {
+                        /// For insert query chose only replicas with write access
+                        if (!check_write_access_to_table.empty())
+                        {
+                            auto query = entry->query("SELECT count(*) > 0 "
```

Comment:
Is it possible, that user does not have access to `information_schema.table_privileges`?